src/GitVersionCore/GitVersionCore.csproj

```diff
@@ -47,6 +47,7 @@
     </Reference>
     <Reference Include="System" />
     <Reference Include="System.Core" />
+    <Reference Include="System.Web" />
```

Comment:
StringFormatWith.cs needs them for the DataBinder class which can throw HttpException.  This is using the code you recommended to do the formatting, which definitely works well.  Another option might be to use C#6 string format inlining e.g. $"{foo} {bar}" instead of string.Format("{0} {1}", foo, bar).  But i wasn't sure you wanted to require C#6 semantics for the code to compile.

try
            {
                if (string.IsNullOrEmpty(format))
                {
                    return (DataBinder.Eval(source, expression) ?? "").ToString();
                }
                return DataBinder.Eval(source, expression, "{0:" + format + "}");
            }
            catch (HttpException)
            {
                throw new FormatException("Failed to format '" + expression + "'.");
            }